### PR TITLE
Use strip_include_prefix and include_prefix to avoid rewriting QUICHE platform include paths

### PR DESCRIPTION
**Pull request by RyanTheOptimist**
Use strip_include_prefix and include_prefix to avoid rewriting QUICHE platform include paths in quiche.genrule_cmd. In the quiche repo, platform APIs include platform impl with the path  "quiche_platform_impl/". In Envoy, the platform impls live in either //source/common/quic/platform/ or //test/common/quic/platform/ and the BUILD files specifies and include_prefix of "quiche_platform_impl".

Risk Level: N/A
Testing: N/A
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

**Comment by RyanTheOptimist** [comment]
@danzh2010 Please take a look. Using Bazel's `include_prefix` and `strip_include_prefix` we are able to get rid of (most of) the include rewriting in quich.gencmd. Once more of the quiche platform refactoring is finished, we'll be able to get rid of more of it.

FWIW, the clang-tidy errors seem to be spurious ala the logging.h tidy errors last week. (I think our clang-tidy handling doesn't understand `strip_include_prefix`)

**Comment by danzh2010** [comment]
> @danzh2010 Please take a look. Using Bazel's `include_prefix` and `strip_include_prefix` we are able to get rid of (most of) the include rewriting in quich.gencmd. Once more of the quiche platform refactoring is finished, we'll be able to get rid of more of it.
> 
> FWIW, the clang-tidy errors seem to be spurious ala the logging.h tidy errors last week. (I think our clang-tidy handling doesn't understand `strip_include_prefix`)

This is a brilliant idea! As this PR removes special handlings in .gencmd, it will greatly simplify Google's envoy import as well.

**Comment by RyanTheOptimist** [comment]
Awesome! Care to approve?

File: source/common/quic/platform/BUILD [hunk 1]
```diff
@@ -41,6 +41,7 @@ envoy_cc_library(
         "abseil_base",
         "abseil_synchronization",
     ],
+    include_prefix = "quiche_platform_impl",
```
[the remark this answers is not in the document]
Comment (RyanTheOptimist) [reply source/common/quic/platform/BUILD:44]: Oh very clever! Done!

**Review by RyanTheOptimist — COMMENTED** [review]
Thanks for the great idea!

File: source/common/quic/platform/BUILD (hunk 1 above)
[the remark this answers is not in the document]
Comment (danzh2010) [reply source/common/quic/platform/BUILD:44]: Not done in source/common/quic/platform/BUILD or test/common/quic/platform/BUILD?

[the remark this answers is not in the document]
Comment (RyanTheOptimist) [reply source/common/quic/platform/BUILD]: Ah, good point. Done!

**Review by danzh2010 — DISMISSED** [review]
Nice!

**Comment by RyanTheOptimist** [comment]
cc: @envoyproxy/dependency-shepherds can you review for the bzl changes?

**Review by htuch — COMMENTED** [review]
/lgtm deps

**Comment by RyanTheOptimist** [comment]
@itayd do you have any idea why repokitteh is saying: no relevant owners for "deps"?

**Comment by RyanTheOptimist** [comment]
> @itayd do you have any idea why repokitteh is saying: no relevant owners for "deps"?

Oh, @htuch figured out that there are no longer any files which needed deps approval. sorry for the noise!

**Review by alyssawilk — COMMENTED** [review]
No objections to this one but can you put a bit more detail in the description about what the functional impact is? And is this the one where we expect to force merge over clang tidy?

**Comment by RyanTheOptimist** [comment]
> No objections to this one but can you put a bit more detail in the description about what the functional impact is?

Done.

> And is this the one where we expect to force merge over clang tidy?

Yup, much like logging.h it seems clang-tidy has issues with strip_include_prefix

**Pull request merged**